frontend/src/Game/components/Game.tsx

```diff
@@ -160,127 +185,94 @@ function SpeedPU(props: { speed: Ref }): ReactElement {
-        }
-
-        clientPlayersPaddle.forEach((value) => {
-          if (value.id === 'left') {
-            setLeftPaddle(value)
-          } else if (value.id === 'right') {
-            setRightPaddle(value)
-          }
-        })
-      }
-    }
-    setTicks(time)
-  }, status.current === EStatus.set)
+  useEffect(() => {
+    socket.on('updateStatus', (serverStatus: EStatus) => {
+      setStatus(serverStatus)
+    })
+  }, [])
 
   return (
     <Col id="centerCol">
-      <SpeedPU speed={speed} />
+      <SpeedPU leftID={props.match.leftPlayer.socketID} status={status} />
       <div id="match">
-        <div id="boardDiv" />
-        <div id="leftScore">{score.current.left}</div>
-        <div id="rightScore">{score.current.right}</div>
-        <div id="countDown">
-          {status.current === EStatus.ready && (
-            <CountDown ticks={ticks} status={status} />
-          )}
-        </div>
-        {status.current === EStatus.play && <Ball ball={ball} />}
-        {status.current === EStatus.set && <Result score={score.current} />}
-        <Paddle paddle={leftPaddle} />
-        <Paddle paddle={rightPaddle} />
+        <div id="boardDiv"></div>
+        <Score
+          left={props.match.leftPlayer.score}
+          right={props.match.rightPlayer.score}
+        />
+        <div id="countDown">{status === EStatus.ready && <CountDown />}</div>
+        {status === EStatus.play && <Ball ball={props.match.ball} />}
+        {status === EStatus.set && (
+          <Result
+            score={{
+              left: props.match.leftPlayer.score,
+              right: props.match.rightPlayer.score
+            }}
+          />
+        )}
+        {status !== EStatus.none && (
+          <Paddles
+            leftSocketID={props.match.leftPlayer.socketID}
+            rightSocketID={props.match.rightPlayer.socketID}
+            leftPaddle={props.match.leftPlayer.paddle}
+            rightPaddle={props.match.rightPlayer.paddle}
+            status={status}
+          />
+        )}
       </div>
     </Col>
   )
 }
 
-function Ready(props: { player: IPlayer; setPlayer: Setter }): ReactElement {
+function Ready(props: { player: IPlayer }): ReactElement {
   const greenButton = 'btn btn-success btn-lg pull bottom'
   const grayButton = 'btn btn-secondary btn-lg pull bottom'
   const [button, setButton] = useState<string>(grayButton)
   const matchState = useLocation().state
 
   function setReady(): void {
-    let isPlayer: boolean
-    if (props.player.side === 'left') isPlayer = leftID === selfID
-    else isPlayer = rightID === selfID
     if (
-      isPlayer &&
+      props.player.socketID === selfID &&
       button === grayButton &&
-      props.player.name === matchState.userName
+      selfName === matchState.userName
     ) {
       setButton(greenButton)
-      props.setPlayer({ ...props.player, ready: true })
-      socket.emit('updatePlayerReady', props.player.side)
+      props.player.ready = true
```

Comment:
setMatchを使わずにマッチのstateを変えているのが少し気になりましたが、ready playerをemittingしているので全てうまくいっているのでしょうか？state setter関数を使わずにstateを変更しない方が良いと思います。emitがあるのでplayer.readyは必要ないように思えるので、もしかしたらIPlayerから削除してもいいかもしれません？あと、よくわからないけど、if(button === gray) setButtonはここに移動できるのでは？

I was a little concerned that the match state is being changed without using setMatch, but I think everything is working well since we are emitting the ready player? I think it's best to not change state without the state setter function. It seems like we don't need player.ready since we have the emit, so maybe it can be removed from IPlayer completely. And I'm not sure, but maybe if(button === gray) setButton can move here?
```suggestion
      if (button === grayButton) setButton(greenButton)
```

Reply:
そうですね。props.player.ready = trueは必要ないです。　対応完了

if(button === gray) setButton　は再レンダリングされた時にgreenButtonにするという役割があります。なのでsetReady functionの中に、この1行を入れてしまうとreadyしているのにgreenButtonではないということが起きるので、このままで良いと思います！

Reply:
そうかもしれないと思っていました。 ありがとうございました！